utils/oscap-oval.c

```diff
@@ -333,148 +334,53 @@ int app_collect_oval(const struct oscap_action *action)
 
 int app_evaluate_oval(const struct oscap_action *action)
 {
-
-	struct oval_definition_model	*def_model = NULL;
-	struct oval_variable_model	*var_model = NULL;
-	struct oval_directives_model	*dir_model = NULL;
-	oval_agent_session_t		*sess      = NULL;
+	struct oval_session *session;
+	oval_result_t eval_result;
 	int ret = OSCAP_ERROR;
+	const char *val;
 
-	struct oscap_source *main_source = oscap_source_new_from_file(action->f_oval);
-	struct ds_sds_session *sds_session = NULL;
-	struct oscap_source *oval_source = NULL;
-
-	if (action->probe_root) {
-		if (setenv("OSCAP_PROBE_ROOT", action->probe_root, 1) != 0) {
-			fprintf(stderr, "Failed to set the OSCAP_PROBE_ROOT environment variable.\n");
-			goto cleanup;
-		}
+	/* create a new OVAL session */
+	if ((session = oval_session_new(action->f_oval)) == NULL) {
+		oscap_print_error();
+		return ret;
 	}
 
-	/* validate inputs */
-	if (action->validate) {
-		if (!valid_inputs(action)) {
-			goto cleanup;
-		}
-	}
+	/* set validation level */
+	val = getenv("OSCAP_FULL_VALIDATION");
+	oval_session_set_validation(session, action->validate, (val != NULL && strncmp(val, "1", 1) == 0) ? true : false);
```

Comment:
This strncmp changes behavior. If you believe it is helpful, please file a separate PR which fixes all occurrences of O_F_V